### DISCUSSION TITLE
Airflow 2.3.3 doesn't retry a task if it externally killed

### DISCUSSION THREAD

**Discussion by nofarm3** [apache/airflow#28794] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

When a task get externally killed, it is marked as Failed even though it can be retried.
We are using kubernetes executor and AWS spots.

### What you think should happen instead

When a task get externally killed (for example by a taken spot), it should put back to retry if it retries did not run out yet.

### How to reproduce

We are using Kubernetes executor and AWS spots.

### Operating System

linux

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Any logs showing this situation ? Can you please be more specific what happens and when ?

**Comment by potiuk** [comment]
I think without more detailed evidences it is extremely difficult to even diagnose if you made a mistake in assessing the situation  or whether this is a real issue. Let me convert it into a discussion until the user provides more evidences allowing to see a reproducile path. We can always convert it into an issue if it turns to be one.
